sunspot/lib/sunspot/admin_session.rb

```diff
@@ -72,18 +72,25 @@ def add_hostname(host)
     # Return all collections. Refreshing every @refresh_every (30.min)
     # Array:: collections
     def collections(force: false)
-      with_cache(force: force, key: 'CACHE_SOLR_COLLECTIONS', default: []) do
+      cs = with_cache(force: force, key: 'CACHE_SOLR_COLLECTIONS', default: []) do
         resp = solr_request('LIST')
+        return [] if resp.nil?
+
         r = resp['collections']
         !r.is_a?(Array) || r.count.zero? ? [] : r
       end
+
+      # belive or not this line is needed only for Ruby 2.3
```

Comment:
instead of commenting, why doesn't explicitelly code it?
like?
`Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.4')`